### PR TITLE
Parsenum

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -25,6 +25,7 @@
  */
 #include "scrypt_platform.h"
 
+#include <math.h>
```
Comment (gperciva) [inline main.c]: I added this for INFINITY, used (at the moment) for `-t maxtime`.  I'm not certain if you want an upper bound on that -- sure, if they wanted to not time out, they'd just omit the `-t` parameter, but the original code didn't impose an upper bound.

**Comment by cperciva** [comment]
Supporting INFINITY is good -- not that we particularly want to support INFINITY itself, but INFINITY is the only upper bound we can specify which will be greater than any non-infinite value. :-)

Is the `(double)` cast necessary though?

**Comment by gperciva** [comment]
Yes; that's why `libcperciva/tests/parsenum/main.c` includes it.  Without the `(double)`, `clang` gives us:

```
main.c:111:34: error: implicit conversion increases floating-point precision:
      'float' to 'double' [-Werror,-Wdouble-promotion]
        TEST4_SUCCESS("inf", double, 0, INFINITY, INFINITY);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
```

`gcc` doesn't complain.  I suppose we could add `-Wno-double-promotion` to `parsenum.h` instead of doing the cast in the `.c` file, though.

**Comment by cperciva** [comment]
Oh, I assumed it was necessary in `TEST4_SUCCESS` due to the test-framework bits, not the parsenum call.  Hmm... is there any reason to not have the explicit `(double)` cast in parsenum itself?

**Comment by gperciva** [comment]
Sketch of a solution in https://github.com/Tarsnap/libcperciva/pull/61.  I'm not wild about one part of it.

**Comment by gperciva** [comment]
Updated PR

**Pull request merged**